### ISSUE TITLE
Factor out MonteCarloIntegration.jl into extension package

### ISSUE THREAD

**Issue by frankier** [SciML/Integrals.jl#207]
**Is your feature request related to a problem? Please describe.**

Currently https://github.com/ranjanan/MonteCarloIntegration.jl depends on an old version of https://github.com/SciML/QuasiMonteCarlo.jl/ . It would be nice if it was possible to use this package and new versions of QuasiMonteCarlo.jl

**Describe the solution you’d like**

Factoring MonteCarloIntegration.jl into an extension package would help. Especially since MCIntegration.jl will be added soon, which could be a reasonable alternative.

**Describe alternatives you’ve considered**

Updating MonteCarloIntegration.jl . However, this has been delayed for quite a while. There have also been other problems like https://github.com/SciML/Integrals.jl/pull/195

I guess it is still maintained but slowly so the extension package approach might make sense for the time being at least. If it ends up being useful in addition to MCIntegration.jl long-term, it may need new maintainer. Otherwise I guess MCIntegration.jl will superceed it.

**Who will do it**

I can make a PR for this change after preliminary approval if but only in case it would be helpful.

**Comment by ChrisRackauckas** [comment]
Yes. It is basically superseded by MCIntegration.jl at this point, so https://github.com/SciML/Integrals.jl/pull/202 is going to move it to an extension and make MCIntegration.jl the new core dep.

**Comment by frankier** [comment]
That makes sense, however just FYI in my opinion it is not yet at feature parity since it doesn't seem to support QMC https://github.com/numericalEFT/MCIntegration.jl/issues/53
